src/async/GenericMuxedInterruptDistributor.cc

```diff
@@ -3,27 +3,718 @@
+     *   - if neither (SIE and CIE) nor IMR are present, or only IER is there, it writes ( ``1<<N``)|(`1<<L`)
+     * to IER and _clears_ with `1<<L`
+     */
+    _activeInterrupts |= mask;
+    try {
+      if(_ierIsReallyImaskr) { // Set IMaskR in the form of IER
+        _ier->accessData(0) = ~_activeInterrupts;
+        _ier->write();
+        // When IMaskR is used, SIE and CIE cannot be defined.
+      }
+      else {
+        if(_haveSieAndCie) { // Set SIE
+          _sie->accessData(0) = _activeInterrupts;
+          _sie->write();
+        }
+        else { // Set IER, which actually is IER and not IMaskR
+          _ier->accessData(0) = _activeInterrupts;
+          _ier->write();
+        }
+      }
+    }
+    catch(ChimeraTK::runtime_error&) {
+    }
+  } // enableInterruptFromMask
+
+  /********************************************************************************************************************/
+  inline void GenericMuxedInterruptDistributor::enableOneInterrupt(uint32_t ithInterrupt) {
+    enableInterruptsFromMask(iToMask(ithInterrupt));
+  }
+
+  /********************************************************************************************************************/
   void GenericMuxedInterruptDistributor::handle(VersionNumber version) {
-    // Stupid testing implementation that always triggers all children
+    try {
+      _isr->read();
+      uint32_t ipr = _activeInterrupts & _isr->accessData(0);
+
+      for(auto const& [i, subDomainWeakPtr] : _subDomains) {
+        if(ipr & iToMask(i)) {
+          if(auto subDomain = subDomainWeakPtr.lock(); subDomain) {
+            //  The weak pointer might have gone.
+            //  TODO FIXME: We need a cleanup function which removes the map entry.
+            //  Otherwise we might be stuck with a bad weak pointer which is tried in each handle() call.
+
+            subDomain->distribute(nullptr, version);
+
+            // Requirement: nested interrupt handlers must clear their active interrupt flag first,
+            // then the parent interrupt flags are cleared.
+            clearOneInterrupt(i);
+          }
+        }
+      } // for
+    }
+    catch(ChimeraTK::runtime_error&) {
+      // There's nothing to do. The transferElement part of _activeInterrupts has already called the backend's setException
+    }
+  } // handle
+
+  /********************************************************************************************************************/
+  std::unique_ptr<GenericMuxedInterruptDistributor> GenericMuxedInterruptDistributor::create(
+      [[maybe_unused]] std::string const& description, boost::shared_ptr<SubDomain<std::nullptr_t>> parent) {
+    /*
+     *  This is a factory function. It parses the json, and calls the constructor.
+     *  It returns an initalized GenericMuxedInterruptDistributor.
+     *  'descriptor' is a JSON snippet containing configuration data
```

Comment:
descriptor->description

Reply:
No, "descriptor" is correct. Like the distributor is the entity doing the distribution, the descriptor is the object doing the description.

Reply:
What I meant is that the comment does not match the argument name

Reply:
Ah, I did not see this from the review page. Code context is missing. I will check.

Reply:
yep, fixed.